### PR TITLE
Add missing setProvider call

### PR DESCRIPTION
**Pull request by KannuSingh**
# Description

For Embedded wallet, on successful connection, provider returns undefined for below code.  
`ProviderUtil.getProvider(CommonConstantsUtil.CHAIN.EVM)`
This PR is to fix this issue.

## Type of change

- [ ] Chore (non-breaking change that addresses non-functional tasks, maintenance, or code quality improvements)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Associated Issues

For Linear issues: Closes APKT-xxx
For GH issues: closes #...

# Showcase (Optional)

If there is a UI change include the screenshots with before and after state.
If new feature is being introduced, include the link to demo recording.

# Checklist

- [ ] Code in this PR is covered by automated tests (Unit tests, E2E tests)
- [ ] My changes generate no new warnings
- [ ] I have reviewed my own code
- [ ] I have filled out all required sections
- [ ] I have tested my changes on the preview link
- [ ] Approver of this PR confirms that the changes are tested on the preview link

### CODE REVIEW

**Comment by enesozturk** [comment]
@KannuSingh can you add changeset file please?

File: packages/adapters/wagmi/src/client.ts [hunk 1]
```diff
@@ -726,6 +726,7 @@ export class WagmiAdapter implements ChainAdapter {
           ])
           if (connector) {
             this.syncConnectedWalletInfo(connector)
+            ProviderUtil.setProvider(this.chainNamespace, await connector.getProvider())
```
Comment (enesozturk) [inline packages/adapters/wagmi/src/client.ts:729]: We might need to call `ProviderUtil.setProviderId` as well?

File: .changeset/spotty-buses-think.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@reown/appkit-adapter-wagmi': patch
+---
+
```
[the remark this answers is not in the document]
Comment (enesozturk) [reply .changeset/spotty-buses-think.md:4]: @KannuSingh

File: packages/adapters/wagmi/src/client.ts (hunk 1 above)
Comment (enesozturk) [reply packages/adapters/wagmi/src/client.ts:729]: @KannuSingh

Comment (KannuSingh) [reply packages/adapters/wagmi/src/client.ts:729]: @enesozturk  what would be the correct way to get `providerId` of connected provider?

Comment (KannuSingh) [reply packages/adapters/wagmi/src/client.ts:729]: Got it, its `connector.id`

File: packages/adapters/wagmi/src/tests/client.test.ts [hunk 3]
```diff
@@ -147,7 +147,11 @@ describe('Wagmi Client', () => {
     it('should sync account correctly when connected', async () => {
       const mockAddress = '0x1234567890123456789012345678901234567890'
       const mockChainId = 1
-      const mockConnector = { id: 'mockConnector', name: 'Mock Connector' }
+      const mockConnector = {
+        id: 'mockConnector',
+        name: 'Mock Connector',
+        getProvider: vi.fn().mockResolvedValue({})
```
[the remark this answers is not in the document]
Comment (KannuSingh) [reply packages/adapters/wagmi/src/tests/client.test.ts:153]: this is an existing test for `syncAccount`, just added `getProvider` on mockConnector bc I added `connector.getProvider` call in the syncAccount

**Review by enesozturk — APPROVED** [review]

**Pull request merged**